x/gamm/pool-models/stableswap/pool.go

```diff
@@ -123,8 +127,14 @@ func (p Pool) getScaledPoolAmts(denoms ...string) ([]sdk.Dec, error) {
 			return []sdk.Dec{}, fmt.Errorf("denom %s does not exist in pool", denom)
 		}
 		scalingFactor := p.GetScalingFactorByLiquidityIndex(liquidityIndex)
+
+		// should not be possible after pool creation checks but we keep this here as a guardrail
+		if int64(scalingFactor) <= 0 {
+			panic("invalid scaling factor encountered when scaling pool assets")
+		}
```

Comment:
Not sure how I feel about these being everywhere, but if you want this lets add it into the get method, since the property is that this should never get into state.